scripts/shared.js

```diff
@@ -762,20 +822,70 @@ export async function getRelatedArticles(article, relatedCount = 5) {
  * @returns {Promise} Resolves when the block is complete.
  */
 export async function buildArticleCardsBlock(articles, addBlockToDom) {
-  const ul = document.createElement('ul');
-  articles.forEach((article) => {
-    const li = document.createElement('li');
-    const articleUrl = `${window.location.protocol}//${window.location.host}${article.path}`;
-    li.innerHTML = `
-      <a href="${articleUrl}" title="${article.title}" aria-label="${article.title}">
-        ${articleUrl}
-      </a>
-    `;
-    ul.append(li);
+  const block = document.createElement('div');
+  block.classList.add('article-cards', 'block');
+  block.setAttribute('data-block-name', 'article-cards');
+  block.innerHTML = '<div><div></div></div>';
+  await buildArticleCardsContent(articles, block.children.item(0).children.item(0));
+  addBlockToDom(block);
+  loadBlock(block);
```

Comment:
@kunwarsaluja I'd be interested in your opinion on this. I'm essentially "manually" building an `article-cards` block, because in many cases we already have a list of article records from the index.

When the `article-cards` block is authored, it consists of a `<ul>` containing URLs to the articles to include in the block:

![Screenshot 2023-10-03 at 4 14 26 PM](https://github.com/hlxsites/channelco-crn-com/assets/5108740/6b6f6c5e-fe81-4bc3-97ac-84d092e73aeb)

The block would then parse out all the URLs, request each article from the index by its path, then build the block structure based on the records.

We already have a list of articles in the category, company, author, and tag templates, so it wouldn't make sense to add the URLs to an article-cards block just to turn around and query each one again. So this is what I came up with to improve that inefficiency. It also fixes the issue where the article URL list was flickering temporarily.

What are your thoughts?

Reply:
@mfrisbey  yea i dont see a problem why cant we do it this way. We’ve done this in other projects too case by case basis to help with not doing repetitive calls back to query index json.

I see the LHS score dipped a bit not sure if this due to this refactoring or another factor (missing content to fill up space to up the CLS) is causing this

Reply:
@kunwarsaluja I think the LHS and performance dip are related. I'm doing some investigation, but I don't think it's directly related to this PR. I'll hold off on merging, and will let you know what I find.